src/chain/electrum.rs

```diff
@@ -42,7 +42,83 @@ const BDK_ELECTRUM_CLIENT_BATCH_SIZE: usize = 5;
 const ELECTRUM_CLIENT_NUM_RETRIES: u8 = 3;
 const ELECTRUM_CLIENT_TIMEOUT_SECS: u8 = 10;
 
-pub(crate) struct ElectrumRuntimeClient {
+pub(super) enum ElectrumRuntimeStatus {
```

Comment:
Maybe the diff would improve a bit if you left this at roughly the same location as where it was.